src/metabase/sync/util.clj

```diff
@@ -135,6 +135,12 @@
     (driver/sync-in-context (driver.u/database->driver database) database
       f)))
 
+(def ^:private exception-classes-not-to-retry
+  [java.net.ConnectException])
+
+(defn- ^:private flatten-exception-causes
+  [e]
+  (when e (mapv #(:type %) (:via (Throwable->map e)))))
```

Comment:
It might be easier just to use iterate and .getCause

We're doing that elsewhere in the codebase. Actually I'm pretty sure there's some exception-chain code already somewhere else, might be worth sticking that in util and using it here too